src/githubToSlackMapping.ts

```diff
@@ -0,0 +1,80 @@
+import {info, warning} from '@actions/core'
+import {OctokitClient} from './github/types'
+import {getSlackUserFromName} from './slack/getSlackUserFromName'
+import {SlackClient} from './slack/SlackClient'
+import {type MessageAuthor} from './slack/types'
+
+/**
+ * Return all GitHub usernames from the specified `github_org`, mapped to their Slack user.
+ *
+ * The users are matched by full name. If a match is not found, the value will be `null`.
+ */
+export async function githubToSlackMapping(
+  octokit: OctokitClient,
+  slack: SlackClient,
+  github_org: string
+): Promise<Record<string, MessageAuthor | null>> {
+  const githubUsers = await fetchGitHubUsers(octokit, github_org)
+  if (githubUsers.length === 0) {
+    warning('No GitHub members found.')
+    return {}
+  }
+
+  info('Fetching Slack users')
+  const slackUsers = await slack.getRealUsers()
+  if (slackUsers.length === 0) {
+    warning('No Slack users found.')
+    return {}
+  }
+
+  const result: Record<string, MessageAuthor | null> = {}
+
+  for (const githubUser of githubUsers) {
+    try {
+      const slackUser = getSlackUserFromName(slackUsers, githubUser.name)
+
+      result[githubUser.login] = {
+        slack_user_id: slackUser.id,
+        username: slackUser.profile.display_name,
+        icon_url: slackUser.profile.image_48
+      }
+    } catch (err) {
+      warning(err instanceof Error ? err.message : String(err))
+
+      result[githubUser.login] = null
+    }
+  }
+
+  return result
+}
```

Comment:
4edb065d384ae1e2e057309345a195e90c99e64a simplified this implementation a bit.

Also notably, I widened the return interface to include `null` values, signifying a Slack user could not be matched. I was thinking that could enable a nice `getMessageAuthor` optimization – essentially skipping the expensive API calls when we already know a user will not match (via `null`) while still falling back to them for a presumably new / unknown user (via `undefined`).

Thoughts on that proposed behavior? I didn't update `getMessageAuthor` yet but happy to do so if you're onboard.

Reply:
Updated `getMessageAuthor` in 4732a6af036898a6dc6f88f5c2162594d80cfc14.